src/ClientResolver.php

```diff
@@ -110,6 +110,13 @@ class ClientResolver
             'fn'      => [__CLASS__, '_apply_credentials'],
             'default' => [CredentialProvider::class, 'defaultProvider'],
         ],
+        'stats' => [
+            'type'  => 'value',
+            'valid' => ['bool', 'array'],
+            'default' => ['retries' => true, 'http' => false],
+            'doc'   => 'Set to true to gather transfer statistics on requests sent. Alternatively, you can provide an associative array with the following keys: retries: (bool) Set to false to disable reporting on retries attempted; http: (bool) Set to false to disable collecting statistics from lower level HTTP adapters (e.g., values returned in GuzzleHttp\TransferStats).',
```

Comment:
I think these should be opt-in by default instead of opt-out by default (meaning things default to off and must be turned on). This allows us to add more ``stats`` features in the future without automatically opting users into each new check. If we keep this opt-out by default, then each time we add a new ``stats`` feature, users will need to update their configuration to opt-out of the new feature.

Presumably, if someone is utilizing the array form of the stats config, they explicitly configured it for a reason.